### PR TITLE
Fix Bugs Introduced by #845 and QuadTree

### PR DESCRIPTION
**Pull request by lzardy**
1. Fixes vision issues
2. Fixes collision issues, specifically ones regarding buildings

### CODE REVIEW

**Comment by lzardy** [comment]
I'd like to note that QuadTree seems to be a volatile addition in its current state.

File: GameServerLib/Content/NavGrid.cs [hunk 1]
```diff
@@ -544,7 +544,7 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
 
         public bool IsAnythingBetween(IGameObject a, IGameObject b)
         {
-            var rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
```
Comment (compsi627) [inline GameServerLib/Content/NavGrid.cs]: Why is that needed? Also, I think that will break the lines come after.

Comment (danil179) [inline GameServerLib/Content/NavGrid.cs:547]: Why is that needed? Also, I think that will break the lines come after.

**Comment by danil179** [comment]
> 
> 
> I'd like to note that QuadTree seems to be a volatile addition in its current state.

I'm not sure. The main problem is that the current lib of this QuadTree seems to be very unreliable, but widely used. I don't know if we do something wrong, or it's just the lib.

File: GameServerLib/Content/NavGrid.cs [hunk 2]
```diff
@@ -544,16 +544,20 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
...
             if (a is IObjBuilding)
             {
+                var rayDist = Math.Sqrt(CastRaySqr(b.GetPosition(), a.GetPosition()));
                 rayDist += a.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
```
Comment (danil179) [inline GameServerLib/Content/NavGrid.cs:551]: Take this line out of the if.

File: GameServerLib/Content/NavGrid.cs [hunk 3]
```diff
@@ -544,16 +544,20 @@ public bool IsAnythingBetween(Vector2 a, Vector2 b)
...
             }
             if (b is IObjBuilding)
             {
+                var rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
+                rayDist = Math.Sqrt(CastRaySqr(a.GetPosition(), b.GetPosition()));
                 rayDist += b.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
```
Comment (danil179) [inline GameServerLib/Content/NavGrid.cs:558]: And this. (Delete one of them)

**Review by danil179 — CHANGES_REQUESTED** [review]
Apart from one comment, it looks good.

**Review by danil179 — APPROVED** [review]
Approved.

**Pull request merged**